dashboard/src/actions/datasetListActions.js

```diff
@@ -16,10 +19,12 @@ export const fetchPublicDatasets = (page) => async (dispatch, getState) => {
     let publicData = [...getState().datasetlist.publicData];
     const params = new URLSearchParams();
     params.append("metadata", "dataset.uploaded");
-    params.append("access", "public");
     params.append("offset", offset);
     params.append("limit", limit);
 
+    if (loggedIn) {
+      params.append("mine", "true");
+    }
```

Comment:
This drops the `mine` filter so that public datasets are returned when you're *not* logged in, but it still means that I can't see public datasets *and* my own datasets at the same time. For example, in the compare/visualize view, to be able to compare my new `fio` run against a run that someone else published.

The downside of simply unconditionally dropping the filter is that the list of public datasets could be immense, and it'll be harder to find your own without the generalized filtering UI we've discussed. A simplification might be to have a toggle switch (maybe only when logged in) to select "only show my results" vs "show all results [that I'm allowed to view]" (which includes both public results and my own private results).

Reply:
I'm starting to really wonder about how all this stuff is going to work if/when the Server has 600,000 public datasets on it.

It occurs to me that one thing we might want to consider is a change of approach in terms of selecting the datasets to compare.  That is, suppose the workflow started by selecting a single dataset, and then allowed the user to add datasets to the comparison:  we could/would/should provide lots of filtering support for finding the initial dataset, and then, once that's selected, we would implicitly filter-out any datasets which weren't comparable.  That is, if you picked a fio dataset, then all the uperf (et al.) datasets would disappear from the the choice for comparisons.  This would ease the user's burden, and it would mean that we wouldn't have to report an error when the user selected a an uncomparable combination (because that wouldn't be possible anymore).

Reply:
> It occurs to me that one thing we might want to consider is a change of approach in terms of selecting the datasets to compare. That is, suppose the workflow started by selecting a single dataset, and then allowed the user to add datasets to the comparison: we could/would/should provide lots of filtering support for finding the initial dataset, and then, once that's selected, we would implicitly filter-out any datasets which weren't comparable. That is, if you picked a fio dataset, then all the uperf (et al.) datasets would disappear from the the choice for comparisons. This would ease the user's burden, and it would mean that we wouldn't have to report an error when the user selected a an uncomparable combination (because that wouldn't be possible anymore).

I think any of this is beyond the scope of this PR, but I think I kinda like the idea of something like this. The Visualization view might start empty, with a selection of filters. (I think the filter model should be universal across all views.) But in the particular case of visualization, at least, one simple filter might be to select from a popup of `endpoints.visualization.benchmarks`, which would filter against `server.benchmark`. So that if you want `fio` runs you could just select that and immediately filter a list of all `fio` runs.

And, yeah; if you start with "ordinary" filtering under "compare" mode, your first selection might implicitly provide a `?filter=server.benchmark:<selected>` against the selected dataset's `server.benchmark` narrowing the view to only comparable results.

Reply:
> I think any of this is beyond the scope of this PR

Yes, quite!  But, I think we need to adopt the mindset ASAP that there will be 600K public datasets.  🥴

Reply:
I'd really like to have better visibility filters here: but I think at the very minimum we should just drop the `?mine` entirely and show "all you're allowed to see". It can potentially be large and unwieldy, but forcing users to log out just to see public results is awkward... and won't be obvious to most users anyway.

If it's not too difficult, I'd love to see at least a three-way control here for "show mine", "show public" and "show all"; but we don't want/need to get *too* fancy in this particular PR and just dropping the `?mine` entirely would be an adequate first step to get this PR finished.